controllers/datadogagent/controller_reconcile_v2.go

```diff
@@ -131,27 +126,36 @@ func (r *Reconciler) reconcileInstanceV2(ctx context.Context, logger logr.Logger
 
 	if requiredComponents.ClusterAgent.IsEnabled() {
 		logger.Info("ClusterAgent enabled")
-		result, err = r.reconcileV2ClusterAgent(logger, features, instance, newStatus)
+		result, err = r.reconcileV2ClusterAgent(logger, features, instance, resourceManagers, newStatus)
 		if utils.ShouldReturn(result, err) {
 			return r.updateStatusIfNeededV2(logger, instance, newStatus, result, err)
 		}
 	}
 
 	if requiredComponents.Agent.IsEnabled() {
 		requiredContainers := requiredComponents.Agent.Containers
-		result, err = r.reconcileV2Agent(logger, features, instance, newStatus, requiredContainers)
+		result, err = r.reconcileV2Agent(logger, features, instance, resourceManagers, newStatus, requiredContainers)
 		if utils.ShouldReturn(result, err) {
 			return r.updateStatusIfNeededV2(logger, instance, newStatus, result, err)
 		}
 	}
 
 	if requiredComponents.ClusterChecksRunner.IsEnabled() {
-		result, err = r.reconcileV2ClusterChecksRunner(logger, features, instance, newStatus)
+		result, err = r.reconcileV2ClusterChecksRunner(logger, features, instance, resourceManagers, newStatus)
 		if utils.ShouldReturn(result, err) {
 			return r.updateStatusIfNeededV2(logger, instance, newStatus, result, err)
 		}
 	}
 
+	// ------------------------------
```

Comment:
I'm not sure about the implications of changing the order of this call. I'll need to take a deeper look at the code.
Why is this needed?

Reply:
The local service and network policy weren't getting created when I tested in a cluster. I think it's because this [Apply call](https://github.com/DataDog/datadog-operator/blob/377f96fe19a6a1fd4a9d54590e814fd1065a7944/controllers/datadogagent/controller_reconcile_v2.go#L121) to create the objects in the store was getting called before the reconcile for each component, which is where the global settings are applied. Moving this section to after the component reconciles allowed the local service and network policy to be created.

Reply:
Thanks for clarifying. It think it should be OK :+1: